File: chapter-05-select/code-java/src/main/java/com/example/demo/repository/JooqSelectRepository.java
package com.example.demo.repository;

import com.example.jooq.tables.pojos.Author;
import com.example.jooq.tables.records.AuthorRecord;
import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.stereotype.Repository;

import java.util.List;

import static com.example.jooq.Tables.AUTHOR;

@Repository
@RequiredArgsConstructor
public class JooqSelectRepository {

    private final DSLContext dsl;

    /**
     * 전체 작가 레코드를 조회합니다.
     */
    public List<AuthorRecord> findAllAuthors() {
        return dsl.selectFrom(AUTHOR)
                  .fetch();
    }

    /**
     * 이름(성) 접두어와 최소 ID 조건으로 작가 POJO 리스트를 조회합니다.
     */
    public List<Author> findAuthorsByLastNameAndId(String namePrefix, Integer minId) {
        return dsl.selectFrom(AUTHOR)
                  .where(AUTHOR.LAST_NAME.like(namePrefix + "%")
                      .and(AUTHOR.ID.gt(minId)))
                  .fetchInto(Author.class);
    }
}

File: chapter-05-select/code-kotlin/src/main/kotlin/com/example/demo/repository/JooqSelectRepository.kt
package com.example.demo.repository

import com.example.jooq.tables.pojos.Author
import com.example.jooq.tables.records.AuthorRecord
import com.example.jooq.tables.references.AUTHOR
import org.jooq.DSLContext
import org.springframework.stereotype.Repository

@Repository
class JooqSelectRepository(
    private val dsl: DSLContext
) {
    /**
     * 전체 작가 레코드를 조회합니다.
     */
    fun findAllAuthors(): List<AuthorRecord> {
        return dsl.selectFrom(AUTHOR)
            .fetch()
    }

    /**
     * 이름(성) 접두어와 최소 ID 조건으로 작가 구조체(POJO/DataClass) 리스트를 조회합니다.
     */
    fun findAuthorsByLastNameAndId(namePrefix: String, minId: Int): List<Author> {
        return dsl.selectFrom(AUTHOR)
            .where(
                AUTHOR.LAST_NAME.like("$namePrefix%")
                    .and(AUTHOR.ID.gt(minId))
            )
            .fetchInto(Author::class.java)
    }
}
